fe/src/main/java/org/apache/doris/http/rest/RestTableSchemaAction.java

```diff
@@ -0,0 +1,131 @@
+    }
+
+    public static void registerAction(ActionController controller) throws IllegalArgException {
+        // the extra `/api` path is so disgusting
+        controller.registerHandler(HttpMethod.GET,
+                "/api/{cluster}/{database}/{table}/_schema", new RestTableSchemaAction(controller));
+    }
+
+
+    @Override
+    protected void executeWithoutPassword(ActionAuthorizationInfo authInfo,
+                                          BaseRequest request, BaseResponse response) throws DdlException {
+        // just allocate 2 slot for top holder map
+        Map<String, Object> resultMap = new HashMap<>(2);
+        String clusterName = request.getSingleParameter("cluster");
+        String dbName = request.getSingleParameter("database");
+        String tableName = request.getSingleParameter("table");
+        try {
+            if (Strings.isNullOrEmpty(clusterName)
+                    || Strings.isNullOrEmpty(dbName)
+                    || Strings.isNullOrEmpty(tableName)) {
+                throw new DorisHttpException(HttpResponseStatus.BAD_REQUEST, "No cluster or database or table selected.");
+            }
+            String fullDbName = ClusterNamespace.getFullName(authInfo.cluster, dbName);
+            // check privilege for select, otherwise return 401 HTTP status
+            checkTblAuth(authInfo, fullDbName, tableName, PrivPredicate.SELECT);
+            Database db = Catalog.getCurrentCatalog().getDb(fullDbName);
+            if (db == null) {
+                throw new DorisHttpException(HttpResponseStatus.NOT_FOUND, "Database [" + dbName + "] " + "does not exists");
+            } else {
+                db.readLock();
+                try {
+                    Table table = db.getTable(tableName);
+                    if (table == null) {
+                        throw new DorisHttpException(HttpResponseStatus.NOT_FOUND, "Table [" + tableName + "] " + "does not exists");
+                    } else {
+                        // just only support OlapTable, ignore others such as ESTable, KuduTable
+                        if (!(table instanceof OlapTable)) {
+                            // Forbidden
+                            throw new DorisHttpException(HttpResponseStatus.FORBIDDEN, "Table [" + tableName + "] "
+                                    + "is not a OlapTable, only support OlapTable currently");
+                        } else {
+                            try {
+                                List<Column> columns = table.getBaseSchema();
+                                Map<String, Map<String, String>> propMap = new HashMap<>(columns.size());
+                                for (Column column : columns) {
+                                    Map<String, String> baseInfo = new HashMap<>(2);
+                                    baseInfo.put("type", column.getType().getPrimitiveType().toString());
+                                    baseInfo.put("comment", column.getComment());
```

Comment:
Do you have considered Decimal type? because it's PrimitiveType does't include it's scalar and precision.